src/components/input_fields/code_field.tsx

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2018 Ultimaker B.V.
+import * as React from 'react';
+import _ = require('lodash');
+import classNames from 'classnames';
+
+// components
+import InputFieldWrapper, {InputFieldProps} from './input_field_wrapper';
+
+/**
+ * The props of the code field.
+ */
+export interface CodeFieldProps extends InputFieldProps {
+    /** Input field value */
+    value: string | null;
+    /** Amount of characters required in the field **/
+    maxLength: number;
```

Comment:
I chose maxLength because it's consistent with the other fields. This should work just like a TextField. It's up to the user of the component to decide whether that's also the required length.